OpenRA.Mods.Common/Traits/Air/FallsToEarth.cs

```diff
@@ -42,18 +42,24 @@ public void RulesetLoaded(Ruleset rules, ActorInfo ai)
 		}
 	}
 
-	public class FallsToEarth : IEffectiveOwner
+	public class FallsToEarth : IEffectiveOwner, INotifyCreated
 	{
+		readonly FallsToEarthInfo info;
 		readonly Player effectiveOwner;
 
 		public FallsToEarth(ActorInitializer init, FallsToEarthInfo info)
 		{
-			init.Self.QueueActivity(false, new FallToEarth(init.Self, info));
+			this.info = info;
 			effectiveOwner = init.Contains<EffectiveOwnerInit>() ? init.Get<EffectiveOwnerInit, Player>() : init.Self.Owner;
 		}
 
 		// We return init.Self.Owner if there's no effective owner
 		bool IEffectiveOwner.Disguised { get { return true; } }
 		Player IEffectiveOwner.Owner { get { return effectiveOwner; } }
+
+		void INotifyCreated.Created(Actor self)
+		{
+			self.QueueActivity(false, new FallToEarth(self, info));
```

Comment:
Shouldn't this be a creation activity instead?

Reply:
This is not an activity like `MoveIntoWorld`, so I don't think so. Additionally, we still have an active `Aircraft` trait here, which means making this a creation activity will conflict.